web/core/components/issues/issue-detail/issue-activity/activity-comment-root.tsx

```diff
@@ -28,24 +31,34 @@ export const IssueActivityCommentRoot: FC<TIssueActivityCommentRoot> = observer(
   const activityComments = getActivityCommentByIssueId(issueId);
 
   if (!activityComments || (activityComments && activityComments.length <= 0)) return <></>;
+
+  const isCommentFilterSelected = selectedFilters.includes(EActivityFilterType.COMMENT);
+  const isActivityFilterSelected = selectedFilters.includes(EActivityFilterType.ACTIVITY);
+
+  const filteredActivityComments = activityComments.filter(
+    (activityComment) =>
+      (activityComment.activity_type === "COMMENT" && isCommentFilterSelected) ||
+      (activityComment.activity_type === "ACTIVITY" && isActivityFilterSelected)
+  );
+
   return (
     <div>
-      {activityComments.map((activityComment, index) =>
+      {filteredActivityComments.map((activityComment, index) =>
         activityComment.activity_type === "COMMENT" ? (
           <IssueCommentCard
             projectId={projectId}
             key={activityComment.id}
             workspaceSlug={workspaceSlug}
             commentId={activityComment.id}
             activityOperations={activityOperations}
-            ends={index === 0 ? "top" : index === activityComments.length - 1 ? "bottom" : undefined}
+            ends={index === 0 ? "top" : index === filteredActivityComments.length - 1 ? "bottom" : undefined}
             showAccessSpecifier={showAccessSpecifier}
             disabled={disabled}
           />
         ) : activityComment.activity_type === "ACTIVITY" ? (
           <IssueActivityList
             activityId={activityComment.id}
-            ends={index === 0 ? "top" : index === activityComments.length - 1 ? "bottom" : undefined}
+            ends={index === 0 ? "top" : index === filteredActivityComments.length - 1 ? "bottom" : undefined}
```

Comment:
What is ends?

Reply:
A ticket has been raised for this WEB-1895. I'll take care of it.